### PR TITLE
Use Slice to encode null flags in blocks

### PR DESCRIPTION
**Pull request by martint**
The getRegion method copies chunks of the underlying boolean array on every call.
By switching it to a slice, we can create a view, instead.

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java [hunk 1]
```diff
@@ -92,7 +89,7 @@ public boolean isFull()
     @Override
     public int getSizeInBytes()
     {
-        long size = getRawSlice().length() + SizeOf.sizeOf(valueIsNull);
+        long size = getRawSlice().length() + valueIsNull.getUnderlyingSlice().length();
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: this is different from what `getRawSlice` is doing, you are counting the entire underlying slice for the null slice

File: presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java [hunk 2]
```diff
@@ -199,7 +193,7 @@ private void entryAdded(boolean isNull)
     @Override
     protected boolean isEntryNull(int position)
     {
-        return valueIsNull[position];
+        return valueIsNull.getUnderlyingSlice().getByte(position) != 0;
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: why call `getUnderlyingSlice` here?

Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: oh I see, this is a buider.  never mind

File: presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java (hunk 1 above)
Comment (martint) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: No, getRawSlice() gets the underlying slice of the sliceOutput for the data. I'm doing the same for valueIsNull (which is now a SliceOutput)

Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: I see this is the builder. never mind

File: presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java (hunk 2 above)
Comment (martint) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java]: It's a SliceOutput

File: presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java [hunk 3]
```diff
@@ -24,9 +24,9 @@
     private final int positionCount;
     private final Slice slice;
     private final int[] offsets;
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java]: you will need a similar fix for offsets

**Comment by dain** [comment]
looks good

**Pull request closed without merging**